internal/ackhandler/sent_packet_history.go

```diff
@@ -9,16 +9,18 @@ import (
 
 type sentPacketHistory struct {
 	packets          []*packet
```

Comment:
I wonder if you would see a perf improvement by using a `[]packet` here. That way when iterating over packets located contiguously in memory.

Reply:
That's a good idea. I was concerned about copying the struct over and over again, but then we're shrinking the size of this struct...

We should benchmark that! I'll open an issue.